File: api/Bussiness/Services/BarcodeService/Dtos/BarcodeDto.cs
using Bussiness.Helper.Enums;
using Bussiness.Services.LoginService.Dtos;
using Bussiness.Services.Stores.Dtos;

namespace Bussiness.Services.BarcodeService.Dtos
{
    public class BarcodeDto
    {
        public int Id { get; set; }
        public LocationDto LocationDto { get; set; }     // Cihaz kullanıcısının koordinatları
        public string Data { get; set; }
        public BarcodeReadEnum? BarcodeReadEnum { get; set; }
        //public string? DeviceBrand { get; set; }
        //public string? DeviceToken { get; set; }
        //public string DeviceModelName { get; set; }
        public LoginDto LoginDto { get; set; } = new LoginDto();
        public StoreDto StoreDto { get; set; } = new StoreDto();          // Mağazanın koordinatları
        public int DeviceId { get; set; }
    }
}

File: ui/Components/Barcode/Dtos/barcodeDto.ts
import { initialLoginDto, LoginDto } from "../../Login/Dtos/LoginDto";
import { initialStoreDto, StoreDto } from "../../Stores/Dtos/storeDto";
import { BarcodeReadEnum } from "../../../Enums/barcodeStatusEnum";
import { initialLocationDto, LocationDto } from "../../../Location/Dtos/locationDto";

export interface BarcodeDto {
    id: number;
    barcodeReadEnum: BarcodeReadEnum | null;
    data: string
    locationDto: LocationDto;
    loginDto: LoginDto
    storeDto: StoreDto
    loading: boolean // profildeki kartın giriş durumun yüklenmesi
    deviceId: number
    qrCodeVisibleState:boolean
}

export const initialBarcodeState: BarcodeDto = {
    id: 0,
    barcodeReadEnum: BarcodeReadEnum.Default,
    locationDto: initialLocationDto,
    data: "",
    loginDto: initialLoginDto,
    storeDto: initialStoreDto,
    loading: false,
    deviceId: 0,
    qrCodeVisibleState: true
}